Fix existing lint issues in ir.py

aorenste opened the pull request:
- Remove stale mypy "type: ignores"
- Made ir.py pass the rest of the lints

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #131846
* #139238
* __->__ #139237



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

aorenste committed:
Fix existing mypy issues

[ghstack-poisoned]

aorenste committed:
Update on "Fix existing lint issues in ir.py"

- Remove stale mypy "type: ignores"
- Made ir.py pass the rest of the lints




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang aakhundov

[ghstack-poisoned]

aorenste committed:
Update on "Fix existing lint issues in ir.py"

- Remove stale mypy "type: ignores"
- Made ir.py pass the rest of the lints




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang aakhundov

[ghstack-poisoned]

Skylion007 commented on torch/utils/benchmark/utils/fuzzer.py:
```diff
@@ -373,7 +373,7 @@ def __init__(
         import numpy as np
         if seed is None:
-            seed = np.random.RandomState().randint(0, 2 ** 32 - 1, dtype=np.int64)
+            seed = np.random.RandomState().randint(0, 2 ** 32 - 1, dtype=np.int64)  # type: ignore[assignment]
```
```suggestion
            seed = int(np.random.RandomState().randint(0, 2 ** 32 - 1, dtype=np.int64))
```

aorenste committed:
Update on "Fix existing lint issues in ir.py"

- Remove stale mypy "type: ignores"
- Made ir.py pass the rest of the lints




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang aakhundov

[ghstack-poisoned]

aorenste committed:
Update on "Fix existing lint issues in ir.py"

- Remove stale mypy "type: ignores"
- Made ir.py pass the rest of the lints




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang aakhundov

[ghstack-poisoned]

aorenste committed:
Update on "Fix existing lint issues in ir.py"

- Remove stale mypy "type: ignores"
- Made ir.py pass the rest of the lints




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang aakhundov

[ghstack-poisoned]

aorenste commented:
@pytorchbot merge
